core/src/main/scala/com/avast/clients/rabbitmq/ConsumerWithCallbackBase.scala

```diff
@@ -90,7 +90,7 @@ abstract class ConsumerWithCallbackBase[F[_]: Effect](channel: ServerChannel,
                                     routingKey: String,
                                     body: Array[Byte])(t: Throwable): F[Unit] = {
     F.delay {
-      logger.error(s"[$name] Error while executing callback, it's probably a BUG", t)
+      logger.error(s"[$name] Error while executing callback for delivery with routing key $routingKey", t)
```

Comment:
Why not combine those two? It was intentionally mentioning that this shouldn't happen.

Reply:
I'm not sure I understand you 🤔 This message can be seen in the log e.g. in the case when we are not able to process the message **now**, so the exception bubbles up to the library, causing `Republish`.

So it IMHO doesn't have to be a BUG (a programmer's error) to reach this code, it can be a _runtime_ problem.

Reply:
If you check where this is used, it's:
1. When the user-provided callback returns an `F` which fails.
2. When the user-provided callback doesn't even return anything and fails with throwing an exception (!!!).

Both these cases are IMHO a bug in the callback, a programmer's error and thus it should be reported as that. The user should always handle all the errors in his code and return `Republish` by himself when needed.

I have absolutely no objections against adding more detailed info though.

Reply:
> The user should always handle all the errors in his code and return Republish by himself when needed.

This is the clash 🎉 We don't use this library in this way. We let the callback fail and rely on the [failureAction](https://github.com/avast/rabbitmq-scala-client/blob/master/core/src/main/scala/com/avast/clients/rabbitmq/configuration.scala#L30) feature (default is `Republish`). This is why this state is not a BUG for us.

Reply:
Ah. OK. It was not meant that way. I would swear that it's explicitly written in doc... but I can find it 😄

Well ok, I can survive if the `F` failure is not logged like that. Change the logging that the "it's a BUG!" message will be written if an exception is thrown and I'll be fine with your change.

Reply:
OK, I've refactored the code to use different messages for both cases.